lib/elixir/test/elixir/task_test.exs

```diff
@@ -272,6 +272,139 @@ defmodule TaskTest do
     end
   end
 
+  @tag timeout: 10
+  describe "await_many/2" do
+    test "returns replies in input order" do
+      refs = [ref_1 = make_ref(), ref_2 = make_ref(), ref_3 = make_ref()]
+      tasks = Enum.map(refs, fn ref -> %Task{ref: ref, owner: self(), pid: nil} end)
+      send(self(), {ref_2, 3})
+      send(self(), {ref_3, 9})
+      send(self(), {ref_1, 1})
+      assert Task.await_many(tasks) == [1, 3, 9]
+    end
+
+    test "returns an empty list immediately" do
+      assert Task.await_many([]) == []
+    end
+
+    test "ignores messages from other processes" do
+      other_ref = make_ref()
+      refs = [ref_1 = make_ref(), ref_2 = make_ref()]
+      tasks = Enum.map(refs, fn ref -> %Task{ref: ref, owner: self(), pid: nil} end)
+      send(self(), {ref_2, :b})
+      send(self(), other_ref)
+      send(self(), {other_ref, :z})
+      send(self(), {:DOWN, other_ref, :process, 1, :goodbye})
+      send(self(), {ref_1, :a})
+      assert Task.await_many(tasks) == [:a, :b]
```

Comment:
You should assert the spurious messages were not consumed. So do things like `assert_received {:DOWN, ^other_ref, :process, 1, :goodbye}` here. Also, please test what happens if you do `send(self(), {ref_2, :b})` and then `send(self(), {ref_2, :c})` (i.e. the `:c` one should stay in the inbox).